src/extension.ts

```diff
@@ -121,6 +121,8 @@ export async function activate(context: vscode.ExtensionContext) {
         )
         ext.sdkClientBuilder = new DefaultAWSClientBuilder(awsContext)
         ext.toolkitClientBuilder = new DefaultToolkitClientBuilder(regionProvider)
+        ext.s3fileViewerManager = new S3FileViewerManager()
+        await ext.s3fileViewerManager.createTemp()
```

Comment:
You're already calling `createTemp` from within the s3FileViewerManager (in `getFile`). Thinking this isn't necessary, or could be moved to the constructor. This also means you can make `createTemp` private in the class.